quit/web/modules/endpoint.py

```diff
@@ -51,36 +51,66 @@ def sparql(branch_or_ref):
+                # Set default
+                if mimetype == '*/*':
+                    mimetype = 'application/sparql-results+xml'
+
+                if mimetype in ['text/html', 'application/xhtml_xml']:
                     results = res.serialize(format='html')
                     response = make_response(render_template(
                         "results.html", results=Markup(results.decode())))
                     response.headers['Content-Type'] = 'text/html'
                     return response
-                elif mimetype in ['application/json', 'application/sparql-results+json']:
+                elif mimetype in ['application/sparql-results+json']:
                     response = make_response(res.serialize(format='json'), 200)
                     response.headers['Content-Type'] = 'application/json'
                     return response
-                elif mimetype in [
-                    'application/rdf+xml', 'application/xml', 'application/sparql-results+xml'
-                ]:
+                elif mimetype in ['application/sparql-results+xml']:
                     response = make_response(res.serialize(format='xml'), 200)
                     response.headers['Content-Type'] = 'application/rdf+xml'
                     return response
                 elif mimetype in ['application/csv', 'text/csv']:
                     response = make_response(res.serialize(format='csv'), 200)
                     response.headers['Content-Type'] = 'text/csv'
                     return response
+                else:
+                    response = make_response('', 406)
+                    return response
+            elif query_type in ['CONSTRUCT', 'DESCRIBE']:
+                res = graph.query(q)
+
+                # Set default
+                if mimetype == '*/*':
+                    mimetype = 'text/turtle'
+
+                if mimetype in ['application/rdf+xml', 'application/xml']:
+                    response = make_response(res.serialize(format='xml'), 200)
+                    response.headers['Content-Type'] = 'application/rdf+xml'
+                    return response
+                elif mimetype in ['text/turtle']:
+                    response = make_response(res.serialize(format='turtle'), 200)
+                    response.headers['Content-Type'] = 'text/turtle'
+                    return response
+                elif mimetype in ['application/trig']:
+                    response = make_response(res.serialize(format='trig'), 200)
+                    response.headers['Content-Type'] = 'application/trig'
+                    return response
+                elif mimetype in ['application/n-triples']:
+                    response = make_response(res.serialize(format='nt'), 200)
+                    response.headers['Content-Type'] = 'application/n-triples'
+                    return response
+                else:
+                    return make_response('', 406)
```

Comment:
Maybe this could be replaced by some dictionary:
mimetype -> serializername

the response header content-type should correstpond to the best matching requested content type which was selected for serialization.